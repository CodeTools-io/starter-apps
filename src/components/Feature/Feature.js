import React from 'react';

export default function Feature({ children, name, ...props }) {
  const key = `REACT_APP_FEATURE_${name}`;
  const isActivated = process?.env?.[key]?.toLowerCase?.() === 'true';

  if (isActivated) {
    return children;
  }

  return null;
}
