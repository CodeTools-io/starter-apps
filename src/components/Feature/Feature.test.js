import React from 'react';
import Feature from './Feature';

describe('Feature', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <Feature name="EXAMPLE_FEATURE">
        <p>Content for a feature that should not be rendered</p>
      </Feature>
    );

    expect(container.child).toMatchSnapshot();
  });
});
