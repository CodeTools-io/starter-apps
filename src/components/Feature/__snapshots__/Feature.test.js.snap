// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature can render 1`] = `undefined`;
